testutil/integration/evmos/factory/factory.go

```diff
@@ -81,21 +92,25 @@ func New(
 	}
 }
 
-// createMsgEthereumTx creates a new MsgEthereumTx with the provided arguments.
-// If any of the arguments are not provided, they will be populated with default values.
-func (tf *IntegrationTxFactory) createMsgEthereumTx(
-	privKey cryptotypes.PrivKey,
-	txArgs evmtypes.EvmTxArgs,
-) (evmtypes.MsgEthereumTx, error) {
-	fromAddr := common.BytesToAddress(privKey.PubKey().Address().Bytes())
-	// Fill TxArgs with default values
-	txArgs, err := tf.populateEvmTxArgs(fromAddr, txArgs)
-	if err != nil {
-		return evmtypes.MsgEthereumTx{}, errorsmod.Wrap(err, "failed to populate tx args")
+// GetEvmTransactionResponseFromTxResult returns the MsgEthereumTxResponse from the provided txResult.
+func (tf *IntegrationTxFactory) GetEvmTransactionResponseFromTxResult(
+	txResult abcitypes.ResponseDeliverTx,
+) (*evmtypes.MsgEthereumTxResponse, error) {
+	var txData sdktypes.TxMsgData
+	if err := tf.ec.Codec.Unmarshal(txResult.Data, &txData); err != nil {
+		return nil, errorsmod.Wrap(err, "failed to unmarshal tx data")
+	}
+
+	if len(txData.MsgResponses) != 1 {
+		return nil, fmt.Errorf("expected 1 message response, got %d", len(txData.MsgResponses))
+	}
+
+	var evmRes evmtypes.MsgEthereumTxResponse
+	if err := proto.Unmarshal(txData.MsgResponses[0].Value, &evmRes); err != nil {
+		return nil, errorsmod.Wrap(err, "failed to unmarshal evm tx response")
```

Comment:
Why not use `evmtypes.DecodeTxResponse` here? AFAICT this is literally the exact same utility (except we're checking for non-zero response in that one and for exactly one in this one).

I think we could adjust the `DecodeTxResponse` util to check for exactly 1 rather - I think it's not possible to batch more than one EthTx within a transaction right?